x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/logic/integrations/install_promotion_rules.ts

```diff
@@ -0,0 +1,121 @@
+import type { EndpointInternalFleetServicesInterface } from '../../../../../endpoint/services/fleet';
+import { AI_FOR_SOC_INTEGRATIONS, PROMOTION_RULE_TAG } from '../../../../../../common/constants';
+
+interface InstallPromotionRulesParams {
+  rulesClient: RulesClient;
+  detectionRulesClient: IDetectionRulesClient;
+  ruleAssetsClient: IPrebuiltRuleAssetsClient;
+  ruleObjectsClient: IPrebuiltRuleObjectsClient;
+  fleetServices: EndpointInternalFleetServicesInterface;
+}
+
+export async function installPromotionRules({
+  rulesClient,
+  detectionRulesClient,
+  ruleAssetsClient,
+  ruleObjectsClient,
+  fleetServices,
+}: InstallPromotionRulesParams): Promise<RuleBootstrapResults> {
+  // Get the list of installed integrations
+  const enabledIntegrations = new Set(
+    (
+      await Promise.all(
+        AI_FOR_SOC_INTEGRATIONS.map(async (integration) => {
+          const installation = await fleetServices.packages.getInstallation(integration);
+          return installation ? integration : [];
+        })
+      )
+    ).flat()
+  );
+
+  const latestRuleAssets = await ruleAssetsClient.fetchLatestAssets();
+
+  const latestPromotionRules = latestRuleAssets.filter((rule) => {
+    // Rule should be tagged as 'Promotion' and should be related to an enabled integration
+    return (
+      (rule.tags ?? []).includes(PROMOTION_RULE_TAG) &&
+      rule.related_integrations?.some((integration) => enabledIntegrations.has(integration.package))
+    );
+  });
+
+  const installedRuleVersions = await ruleObjectsClient.fetchInstalledRuleVersions();
+  const installedRuleVersionsMap = new Map(
+    installedRuleVersions.map((version) => [version.rule_id, version])
+  );
+
+  const promotionRulesToInstall = latestPromotionRules.filter(({ rule_id: ruleId }) => {
+    return !installedRuleVersionsMap.has(ruleId);
+  });
+  const { results: installationResults, errors: installationErrors } = await createPrebuiltRules(
+    detectionRulesClient,
+    promotionRulesToInstall.map((asset) => ({
+      ...asset,
+      enabled: true,
+    }))
+  );
+
+  const promotionRulesToUpgrade = latestPromotionRules.filter(({ rule_id: ruleId, version }) => {
+    const installedVersion = installedRuleVersionsMap.get(ruleId);
+    return installedVersion && installedVersion.version < version;
+  });
+  const { results: upgradeResults, errors: upgradeErrors } = await upgradePrebuiltRules(
+    detectionRulesClient,
+    promotionRulesToUpgrade
+  );
+
+  // Cleanup any unknown rules, we don't allow users to install any detection
+  // rules that are not part of the enabled integrations, although that is not
+  // enforced via the API
+  const rulesToDelete = installedRuleVersions.filter(({ rule_id: ruleId }) => {
+    return !latestPromotionRules.some((rule) => rule.rule_id === ruleId);
+  });
+  if (rulesToDelete.length > 0) {
+    await rulesClient.bulkDeleteRules({
+      ids: rulesToDelete.map(({ id }) => id),
+    });
+  }
```

Comment:
I don't think we should mix this logic into the installation route, it's more likely to cause confusion than help IMO. If we decide this licensing is important to validate we should add it to the rule CRUD APIs and perhaps check at rule execution time.

Reply:
We’ll definitely need a more robust solution to disallow CRUD operations for certain rules or rule types if we decide to continue using this approach. That will likely require changes to Alerting RBAC. But for the MVP stage, I think it's fine to cut corners and go with something simple like rule deletion. For now, I'd prefer to have the AI4SOC logic concentrated in one place so that it can be easily rewritten and replaced with something more reliable in the future.